app/models/teacher.rb

```diff
@@ -56,6 +56,8 @@ class Teacher < ApplicationRecord
   validates :api_id, uniqueness: { case_sensitive: false, message: "API id already exists for another teacher" }
   validates :api_ect_training_record_id, uniqueness: { case_sensitive: false, message: "API ect training record id already exists for another teacher" }
   validates :api_mentor_training_record_id, uniqueness: { case_sensitive: false, message: "API mentor training record id already exists for another teacher" }
+  validates :first_became_eligible_for_ect_training_at, immutable_once_set: true
+  validates :first_became_eligible_for_mentor_training_at, immutable_once_set: true
```

Comment:
It feels to me like we're bringing in quite a big dependency here and only using a tiny bit of its functionality.

I think we can _probably_ achieve this with something like:

```ruby
validate :first_became_eligible_for_ect_training_at_does_not_change, on: :update

def first_became_eligible_for_ect_training_at_does_not_change
  if first_became_eligible_for_ect_training_at_changed?
    errors.add(:first_became_eligible_for_ect_training_at, "Not allowed ...")
  end
end
```